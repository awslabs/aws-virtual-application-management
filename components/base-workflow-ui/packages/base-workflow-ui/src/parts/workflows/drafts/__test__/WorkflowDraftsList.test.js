/*
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://aws.amazon.com/apache2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'mobx-react';
import WorkflowDraftsList from '../WorkflowDraftsList';

jest.mock('../../../workflow-common/drafts/WorkflowCommonDraftCard');
jest.mock('../../../workflow-common/ProgressPlaceholder');
jest.mock('../CreateWorkflowDraft');
jest.mock('../../../workflow-templates/WorkflowTemplateCardTabs');

describe('WorkflowDraftsList', () => {
  const workflowDraftsStore = { ready: 'store-ready', load: jest.fn(), startHeartbeat: jest.fn() };
  const stepTemplatesStore = { ready: 'store-ready', load: jest.fn(), startHeartbeat: jest.fn() };
  it('WorkflowDraftsList should renders correctly', () => {
    expect(
      renderer
        .create(
          <Provider workflowDraftsStore={workflowDraftsStore} stepTemplatesStore={stepTemplatesStore}>
            <WorkflowDraftsList.WrappedComponent
              workflowDraftsStore={workflowDraftsStore}
              stepTemplatesStore={stepTemplatesStore}
            />
          </Provider>,
        )
        .toJSON(),
    ).toMatchSnapshot();
  });
});
