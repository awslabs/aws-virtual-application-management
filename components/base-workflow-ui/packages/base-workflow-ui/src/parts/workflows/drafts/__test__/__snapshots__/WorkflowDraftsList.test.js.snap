// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkflowDraftsList WorkflowDraftsList should renders correctly 1`] = `
<div
  className="mb4"
>
  <div
    className="mb3 flex"
  >
    <h3
      className="ui header color-grey mt1 mb0 flex-auto"
    >
      <i
        aria-hidden="true"
        className="edit outline icon align-top"
        onClick={[Function]}
      />
      <div
        className="content left-align"
      >
        Workflow Drafts
      </div>
    </h3>
    <div>
      <button
        className="ui blue basic button"
        onClick={[Function]}
      >
        Create Draft
      </button>
    </div>
  </div>
  <div />
</div>
`;
